commands/login.go

```diff
@@ -221,16 +248,42 @@ func login(ctx context.Context, provider client.OpenIdProvider) (*LoginCmd, erro
 		return nil, fmt.Errorf("failed to generate SSH cert: %w", err)
 	}
 
-	// Write ssh secret key and public key to filesystem
-	if err := writeKeysToSSHDir(seckeySshPem, certBytes); err != nil {
-		return nil, fmt.Errorf("failed to write SSH keys to filesystem: %w", err)
-	}
-
 	idStr, err := IdentityString(*pkt)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse ID Token: %w", err)
 	}
-	fmt.Printf("Keys generated for identity\n%s\n", idStr)
+
+	if _, ok := HasSSHAgent(); ok {
+		pubkey, _, _, _, err := ssh.ParseAuthorizedKey(certBytes)
+		if err != nil {
+			return nil, err
+		}
+		cert, ok := pubkey.(*ssh.Certificate)
+		if !ok {
+			return nil, fmt.Errorf("failed to cast to certificate")
+		}
+
+		lifetime, err := getExpiration(pkt.Payload)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := sshagent.Add(agent.AddedKey{
+			PrivateKey:   signer,
+			Comment:      "openpubkey key",
+			LifetimeSecs: uint32(lifetime.Seconds()),
+			Certificate:  cert,
+		}); err != nil {
```

Comment:
Is there a danger here that a user runs `opkssh login` 50 times in a row for one user account and then runs `opkssh login` again for a different email account, but ssh can only send 7 or so certs and so will never send the actual cert needed?

Reply:
Hmmm, right. Sure. That could be an issue, also maybe if the refresh goes haywire?

But I'm not sure if we should optimize for that error case right out of the gate, or if we should wait a bit and see if it could become a problem.

Reply:
That's the right call